hcl2helper/values.go

```diff
@@ -0,0 +1,131 @@
+package hcl2helper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/hashicorp/packer-plugin-sdk/template/config"
+	"github.com/mitchellh/mapstructure"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+// UnknownVariableValue is a sentinel value that can be used
+// to denote that the value of a variable is unknown at this time.
+// RawConfig uses this information to build up data about
+// unknown keys.
+const UnknownVariableValue = "74D93920-ED26-11E3-AC10-0800200C9A66"
+
+// HCL2ValueFromConfigValue takes a value turns it into
+// a cty.Value so it can be used within, for example, an HCL2 EvalContext.
+func HCL2ValueFromConfigValue(v interface{}) cty.Value {
```

Comment:
Thought of changing this to `CTYValueFromConfigValue` but I didn't like it so I kept the HCL2 to keep it more clear.  `hcl2helper.HCL2ValueFromConfigValue` doesn't sound repeatedly like `hcl2.HCL2ValueFromConfigValue`